sqlglot/expressions.py

```diff
@@ -4186,19 +4308,20 @@ def delete(table, where=None, dialect=None, **opts) -> Delete:
         'DELETE FROM my_table WHERE id > 1'
 
     Args:
-        where (str|Condition): sql conditional parsed into a WHERE statement
+        where (str|Expression): sql conditional parsed into a WHERE statement
+        returning (str|Expression): sql conditional parsed into a RETURNING statement
         dialect (str): the dialect used to parse the input expressions.
         **opts: other options to use to parse the input expressions.
 
     Returns:
         Delete: the syntax tree for the DELETE statement.
     """
-    return Delete(
-        this=maybe_parse(table, into=Table, dialect=dialect, **opts),
-        where=Where(this=where)
-        if isinstance(where, Condition)
-        else maybe_parse(where, into=Where, dialect=dialect, prefix="WHERE", **opts),
-    )
+    delete_expr = Delete().delete(table, dialect=dialect, **opts)
```

Comment:
I think we can do `copy=False` here because the instance created by `Delete()` becomes garbage.